src/Grpc.AspNetCore.Server.GrpcClient/DefaultGrpcClientFactory.cs

```diff
@@ -0,0 +1,55 @@
+﻿#region Copyright notice and license
+
+// Copyright 2019 The gRPC Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#endregion
+
+using Grpc.Core;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Options;
+using System;
+using System.Net.Http;
+
+namespace Grpc.AspNetCore.Server.GrpcClient
+{
+    internal class DefaultGrpcClientFactory : IGrpcClientFactory
```

Comment:
I was going to complain about this because of parallel abstractions - but I think it seems fine since a Grpc client already is a different type from httpclient.